### PR TITLE
feat: allow configurable token vending machine (permission scope and expiry)

### PR DESCRIPTION
**Pull request by anitarua**
Addresses the first part of [#385](https://github.com/momentohq/dev-eco-issue-tracker/issues/385) by creating a new config file where users can specify token permissions and expiry before deploying the token vending machine. Instructions are in the README and examples are in the `config.ts` file.

The TVM CDK stack can be successfully deployed and a locally running vite-chat-app will get a new token from the API gateway endpoint. However, the app was expecting a string rather than a JSON object (which is what the TVM was providing) so I made a change there to accept JSON instead. The app was also using the wrong environment variable name for the cache name so I fixed that as well. I also had to include the CORS header in the Lambda response so that the app would connect to it.

### CODE REVIEW

File: examples/web/vite-chat-app/src/App.tsx
Comment (bruuuuuuuce) [inline examples/web/vite-chat-app/src/App.tsx:101]: thanks for fixing this!

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Comment by anitarua** [comment]
Since the configuration file is just two variables that have to be set, I could implement the same approach with the NextJS app and bundle it into this PR as well if that makes sense to do.

File: examples/nodejs/token-vending-machine/README.md
Comment (cprice404) [inline examples/nodejs/token-vending-machine/README.md:26]: Can we add one more sentence here to give it a bit more color?

something like "For example, you can restrict the permissions for these browser tokens so that they have read-only access or read-write access, and you can also restrict them to specific caches or topics."

Comment (cprice404) [inline examples/nodejs/token-vending-machine/README.md:26]:
```suggestion
Before you deploy the token vending machine, you will need to configure the scope of permissions and the expiry duration for the tokens that it will vend.
```

File: examples/nodejs/token-vending-machine/lambda/config.ts
Comment (cprice404) [inline examples/nodejs/token-vending-machine/lambda/config.ts:24]: i love that you used the permalink for the gh link instead of HEAD :)

but I think for these readme's let's try to steer people toward the dev docs when we can.  so for this one we can use:

https://docs.momentohq.com/develop/sdks/web#credentials-for-browsers

File: examples/nodejs/token-vending-machine/lambda/config.ts
Comment (cprice404) [inline examples/nodejs/token-vending-machine/lambda/config.ts:42]: for this one, maybe:

https://docs.momentohq.com/develop/api-reference/auth-tokens#generateauthtoken-api ?

or feel free to poke around the dev docs site and see if you find something better. or add something better to the dev docs if you feel like there's nothing good enough there yet :) but i think i'd be good with the link above.

File: examples/nodejs/token-vending-machine/lambda/handler.ts
Comment (cprice404) [inline examples/nodejs/token-vending-machine/lambda/handler.ts:22]: This was a great find/fix, thanks.

**Review by cprice404 — COMMENTED** [review]
awesome job. I just had a couple of nits about the readme text but otherwise 🚢 

if I'm not around when you update the PR, feel free to merge it once @bruuuuuuuce approves it again.

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**
